modules/internal/ChapelArray.chpl

```diff
@@ -1790,6 +1857,24 @@ module ChapelArray {
     }
   }  // record _array
 
+  //
+  // A helper function to check array equality (== on arrays promotes
+  // to an array of booleans)
+  //
+  proc _array.equals(that: _array) {
+    if this.rank != that.rank then
+      return false;
+    //    writeln("ranks match!");
+    for d in 1..this.rank do
+      if this.domain.dim(d).size != that.domain.dim(d).size then
+        return false;
+    //    writeln("sizes match");
+    //    writeln("this == that is: ", this == that);
+    return && reduce (this == that);
```

Comment:
Thinking about this some more this morning, and what the most common cases were likely to be that relied upon it (using the identical array of target locales) made me realize that I should add a quick exit to this routine that returns true if this._value == that._value (and, as I type this, I'm realizing I should do the same for the domain map's equal.